pennylane/transforms/hamiltonian_expand.py

```diff
@@ -106,27 +122,38 @@ def hamiltonian_expand(tape, group=True):
             "Passed tape must end in `qml.expval(H)`, where H is of type `qml.Hamiltonian`"
         )
 
-    if group:
-        hamiltonian.simplify()
-        return qml.transforms.measurement_grouping(tape, hamiltonian.ops, hamiltonian.coeffs)
-
-    # create tapes that measure the Pauli-words in the Hamiltonian
-    tapes = []
-    for ob in hamiltonian.ops:
-        # we need to create a new tape here, because
-        # updating metadata of a copied tape is error-prone
-        # when the observables were changed
-        with tape.__class__() as new_tape:
-            for op in tape.operations:
-                qml.apply(op)
-            qml.expval(ob)
-        tapes.append(new_tape)
-
-    # create processing function that performs linear recombination
+    if group or hamiltonian.grouping_indices is not None:
+        # use groups of observables if available or explicitly requested
+        coeffs, obs_groupings = hamiltonian.get_grouping()
+
+        tapes = []
+        for obs in obs_groupings:
+
+            with tape.__class__() as new_tape:
+                for op in tape.operations:
+                    op.queue()
+
+                for o in obs:
+                    qml.expval(o)
+
+            new_tape = new_tape.expand(stop_at=lambda obj: True)
+            tapes.append(new_tape)
+    else:
+        coeffs = hamiltonian.coeffs
+
+        tapes = []
+        for o in hamiltonian.ops:
+            with tape.__class__() as new_tape:
+                for op in tape.operations:
+                    op.queue()
+                qml.expval(o)
+
+            tapes.append(new_tape)
```

Comment:
@josh146 this would be the new structure

Reply:
Nice, from two loops over the Hamiltonian down to 1!